src/api/spec/controllers/attribute_controller_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'rails_helper'
+
+RSpec.describe AttributeController, type: :controller do
+  render_views
+
+  describe '#update' do
+    let!(:admin) { create(:admin_user) }
+    let(:user1) { create(:confirmed_user) }
+    let!(:namespace) { create(:attrib_namespace, name: 'TEST') }
+    let(:attribute) { create(:attrib_type, attrib_namespace: namespace) }
+    let(:dummy) { 'Dummy' }
+
+    let(:data1) do
+      "<definition namespace='#{namespace.name}' name='#{dummy}'>
+        <count>2</count>
+        <default>
+          <value>A</value>
+          <value>B</value>
+        </default>
+        <allowed>
+          <value>A</value>
+          <value>B</value>
+          <value>C</value>
+        </allowed>
+        <modifiable_by user='#{user1.login}'/>
+      </definition>"
+    end
+
+    let(:data2) do
+      "<definition namespace='#{namespace.name}' name='#{dummy}'>
+      <count>2</count>
+      <default>
+        <value>A</value>
+        <value>B</value>
+      </default>
+      <allowed>
+        <value>A</value>
+        <value>B</value>
+      </allowed>
+      <modifiable_by user='#{user1.login}'/>
+    </definition>"
+    end
+
+    it 'will require auth' do
+      post :update, body: data1, format: :xml, params: { namespace: namespace.name, name: dummy }
+      expect(response.status).to eq(401)
+    end
+    it 'will require permissions' do
```

Comment:
This is already tested in `app/policies/attrib_policy.rb`

Reply:
ok, fair enough